ipatests/test_webui/test_selfservice.py

```diff
@@ -287,4 +287,10 @@ def test_reset_sshkey_permsission(self):
         s = "div[name='ipasshpubkey'].multivalued-widget"
         facet = self.get_facet()
         btn = self.find(s, By.CSS_SELECTOR, facet, strict=True)
-        assert btn.is_displayed is not True
+        btn_is_displayed = btn.is_displayed
+
+        # logout needs to prevent using this unpermitted user in other tests
+        try:
+            assert btn_is_displayed is not True
+        finally:
+            self.logout()
```

Comment:
It would be better to add something like `driver.delete_all_cookies()` to `UI_driver.teardown` method.
Also It is good idea about decorator, but we can use it for **login**, not for **logout**. Because we use `init_app()` method for this purpose and it is unclear. I'll give you example:
``` python
@login.as_admin
def test_crud(self):
    self.basic_crud(ENTITY, self.data)
```
or
``` python
@login.as_('some_prepared_user')
def test_some_permissions(self):
    # some code
```